port addcdiv to opinfo

albanD opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55547 move topk to use OpInfo
* #55535 move logaddexp{2} to opinfo
* **#55518 port addcdiv to opinfo**
* #55517 port addcmul to OpInfo

Differential Revision: [D27649411](https://our.internmc.facebook.com/intern/diff/D27649411)

albanD committed:
port addcdiv to opinfo

[ghstack-poisoned]

albanD committed:
Update on "port addcdiv to opinfo"




[ghstack-poisoned]

imaginary-person commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -2508,7 +2508,15 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
+           dtypesIfCUDA=floating_and_complex_types_and(torch.float16, torch.bfloat16),
+           skips=(
+               SkipInfo('TestCommon', 'test_variant_consistency_eager'),
+               SkipInfo('TestGradients', 'test_inplace_grad'),
```
Suggestion:

addcdiv_ doesn't have a corresponding enttry in https://github.com/pytorch/pytorch/blob/master/tools/autograd/derivatives.yaml.
So that means inplace autograd for it isn't supported.

You'd have to add `supports_inplace_autograd=False`, instead of skipping these 2 tests.
`test_variant_consistency_eager` might pass after setting `supports_inplace_autograd=False`, as without it, it also tries to compute backward of the inplace invariant, and that might be why it failed locally for you.

mruberry replied:
Your analysis is good, @imaginary-person, but test_variant_consistency_eager will probably still fail because of the inplace broadcasting the sample inputs want. I made a comment on the previous PR in the stack. Keeping the skip for now is OK (but please add a comment, @albanD, as described in the previous PR!)

imaginary-person commented on test/test_torch.py:
```diff
@@ -7072,10 +7072,7 @@ def tmp(dtype, device):
         _cpu_types, True, [tf32_on_and_off(0.05), _wrap_warn_once("This overload of baddbmm_? is deprecated")]),
     ('bmm', '', _small_3d, lambda t, d: [_small_3d(t, d)],
         1e-5, 1e-5, 1e-5, _float_types_no_half, _cpu_types, False),
-    ('addcdiv', '', _small_2d,
```
Thanks, because of this, I just recollected that I'd have to remove these entries for some other ops!

mruberry reviewed (approved):
Awesome! See comments on previous PR. This can inherit two of the suggestions, I think: adding a comment for the test_variant_consistency_eager skip and updating the metadata to avoid the gradient test skips (as @imaginary-person also suggests).

albanD committed:
Update on "port addcdiv to opinfo"




[ghstack-poisoned]

albanD committed:
Update on "port addcdiv to opinfo"


Differential Revision: [D27649411](https://our.internmc.facebook.com/intern/diff/D27649411)

[ghstack-poisoned]

albanD committed:
Update on "port addcdiv to opinfo"


Differential Revision: [D27649411](https://our.internmc.facebook.com/intern/diff/D27649411)

[ghstack-poisoned]

albanD committed:
Update on "port addcdiv to opinfo"


Differential Revision: [D27649411](https://our.internmc.facebook.com/intern/diff/D27649411)

[ghstack-poisoned]
